src/crypto/base.h

```diff
@@ -56,18 +55,20 @@ auto operator<=>(const T& left, const T& right) {
 }
 
 template <hash_hex_comparable T>
-std::string to_hex_string(const T& val) {
-    return "<{}>"_format(tools::type_to_hex(val));
-}
-
-template <hash_hex_comparable T>
-    requires(
-            std::is_standard_layout_v<T> && sizeof(T) >= sizeof(size_t) &&
-            alignof(T) >= sizeof(size_t))
+    requires(std::is_standard_layout_v<T> && sizeof(T) >= sizeof(size_t))
 struct raw_hasher {
-    size_t operator()(const T& val) const { return *reinterpret_cast<const size_t*>(val.data()); }
+    size_t operator()(const T& val) const {
+        if constexpr (alignof(T) >= sizeof(size_t))
+            return *reinterpret_cast<const size_t*>(val.data());
```

Comment:
Smells like UB? Original backing memory (`unsigned char` essentially) was not declared as `size_t`. Just defaulting to `memcpy` seems fine.